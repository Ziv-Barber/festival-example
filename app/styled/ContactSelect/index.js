// @flow

import React from 'react' // eslint-disable-line no-unused-vars

// Other components that we need:

// Internationalization (i18n):
// import { FormattedMessage } from 'react-intl'
// import createMessages from './messages'

// Material-ui stuff:
// import MuiPaper from '@material-ui/core/Paper'
// import MuiTypography from '@material-ui/core/Typography'
// import MuiGrid from '@material-ui/core/Grid'

// Our types:
import type { ContactSelectPropsType } from './index.types'

/**
 * ContactSelect react component.
 * @reactProps {string} text Parameters for the component.
 * @return New element.
 */
const ContactSelect = (props: ContactSelectPropsType) => (
  <React.Fragment>
    <span>{props.text}</span>
  </React.Fragment>
)

export default ContactSelect

// In case that someone need our types:
export type { ContactSelectPropsType as Props }
