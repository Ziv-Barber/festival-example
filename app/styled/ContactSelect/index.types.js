// @flow

export type ContactSelectPropsType = {
  text: string
}
