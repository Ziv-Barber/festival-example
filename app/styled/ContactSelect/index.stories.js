import React from 'react' // eslint-disable-line no-unused-vars
import { storiesOf } from '@storybook/react'
import ContactSelect from './index'

storiesOf('ContactSelect', module).add(
  'basic',
  () => <ContactSelect text="Basic example" />,
  { notes: { markdown: 'This is an example to the ContactSelect component.' } }
)
