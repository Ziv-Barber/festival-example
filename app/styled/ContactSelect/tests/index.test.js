// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import { render, cleanup } from 'react-testing-library'

import ContactSelect from '../index'

// For flow:
let { describe, it, afterEach, expect } = global

/** @test {ContactSelect} */
describe('<ContactSelect />', () => {
  afterEach(cleanup)

  /** @test {ContactSelect} */
  it('should match snapshort', () => {
    const { container } = render(<ContactSelect text="This is a text" />)
    expect(container).toMatchSnapshot()
  })

  /** @test {ContactSelect} */
  it('should render', () => {
    const { getByText } = render(<ContactSelect text="This is a text" />)
    expect(getByText(/^This is a text/)).not.toBeNull()
  })
})
