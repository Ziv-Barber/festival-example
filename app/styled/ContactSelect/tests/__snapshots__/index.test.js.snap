// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactSelect /> should match snapshort 1`] = `
<div>
  <span>
    This is a text
  </span>
</div>
`;
